### PR TITLE
Docker container for spinmob

### PR DESCRIPTION
**Pull request by MilesCranmer**
Hi,

If you are interested, I made a docker container for spinmob. https://docs.docker.com/engine/

I'm new to Docker myself, but have started to really like it.

Docker is basically virtual machine for software build systems. It also gets native performance which is nice. It is useful for:
- Running codes in identical build environments on different systems (good for reproducibility)
- Painless installation of nearly anything
- Trying out an app/module/library quickly without polluting your build environment
- Continuous integration (e.g., travis has docker ready to go)

It's being used a lot in different software communities, and now quite often for actual production systems (e.g., all of Spotify's servers run on Docker). It's good because you can build a Docker "image," and then just run that image on any system that runs Docker, no matter its build environment. In that sense it's just like virtual machines.

This "Dockerfile" is used to build the docker container. I have it set to use Anaconda 3 as its base layer (which uses a Debian image as its base layer), which gives the container > 100 useful python modules: https://hub.docker.com/r/continuumio/anaconda3/

Setting up the display was a bit tricky so I also have my script for running it on Mac. It's painless to run an ipython notebook from within a Docker container (and talk to it from your normal browser), but for systems that use a full X11 GUI (which I think it required for spinmob?), it's a bit complicated. Moreso on a Mac. 

## Usage
To build (In spinmob directory):
```bash
docker build -t spinmob .
```

To run
```bash
./mac_launch_docker.sh
```

Inside, to e.g., set up a jupyter notebook:
```bash
jupyter notebook --ip="*" --port 8888
```

Then you can access it from your browser at localhost:8888, and have access to spinmob and all the modules included in anaconda3. It's quite painless which is nice. I still haven't figured out why spinmob requires access to my display on import (matplotlib does not, I can run it within the anaconda3 without the complicated launch script), but this should work for Mac. 

Let me know what you think.
Cheers,

Miles

### CODE REVIEW

File: mac_launch_docker.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#Credit to: http://stackoverflow.com/a/36190462/2689923
+
+#Requires Xquartz and socat installed!
+
+#Define what we are running here.
+CONTAINER=spinmob
+COMMAND=/bin/bash
+NIC=en0
+
+# Grab the ip address of this box
```
Comment (jayich) [inline mac_launch_docker.sh]: # comment looks better than 
#comment

I don't know what the bash standard is.

**Comment by jayich** [comment]
@MilesCranmer , thank you so much. This is great.

File: mac_launch_docker.sh (hunk 1 above)
Comment (MilesCranmer) [inline mac_launch_docker.sh]: Good point. I'll edit them.

**Comment by jayich** [comment]
@jaxankey , looks good to me.

**Comment by MilesCranmer** [comment]
Update: I added an equivalent docker launcher for linux. It works for me. Same commands to build and run (but use `./linux_launch_docker.sh`). It uses `$HOME/Documents/workspace:/workspace` as its working directory by default (as in, `/workspace` inside the container gets mapped to `$HOME/Documents/workspace` outside the container).

**Comment by jayich** [comment]
LGTM.

**Comment by jayich** [comment]
@correlator docker check?

**Comment by jayich** [comment]
@correlator , can you check spinmob with docker?

**Comment by jaxankey** [comment]
Probably too much to maintain. I wouldn't object if someone did this on the side, of course.

**Pull request closed without merging**